cloudant-sync-datastore-core/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -117,7 +117,12 @@ public HttpConnection setNumberOfRetries(int numberOfRetries){
      * @return an {@link HttpConnection} for method chaining 
      */
     public HttpConnection setRequestBody(final String input) {
-        this.input = new ByteArrayInputStream(input.getBytes());
+        this.input = new InputStreamGenerator() {
+            @Override
+            public InputStream getInputStream() {
+                return new ByteArrayInputStream(input.getBytes());
+            }
+        };
         // input is in bytes, not characters
         this.inputLength = input.getBytes().length;
```

Comment:
The charset might have an impact on this size, so so should probably be specified again.